optionally keep xml attributes

tam5 opened the pull request:
Closes #150.

From Vim Surround for reference:
> If t or < is used, Vim prompts for an HTML/XML tag to insert.  You may specify
> attributes here and they will be stripped from the closing tag. **If replacing a
> tag, its attributes are kept in the new tag. End your input with > to discard
> the those attributes.** If \<C-T> is used, the tags will appear on lines by
> themselves.

ninrod commented:
Thanks. It seems that your pull breaks the repeat functionality for tags.

tam5 commented:
Yep, looking into it

tam5 commented:
Actually, I am not able to reproduce the error, do you mind having a look if you can reproduce?

ninrod commented:
are you able to run the tests locally?

tam5 commented:
Yes, they all pass. I've also run them in fresh docker container just to be sure, and they passed there as well. Do you mind triggering a rebuild on travis just for a sanity check?

tam5 commented:
got it now, never mind

ninrod commented:
Ok I'll try to run them later today

ninrod commented:
Ok, the tests are passing, the functionality is working. But I'm getting an error on `.` repeating a `ysiwt<a>W.` operation on a buffer that has: 

`tag1 tag2`

the result should be: 

`<a>tag1</a> <a>tag2</a>`

And it is. But I get this ugly note on the minibuffer: `Error running timer ‘exit-minibuffer’: (no-catch exit nil)`

NOTE: I don't the error if I I issue: `ysiwt<a[HIT ENTER]W.` Maybe that's a clue for you

![error-running-timer](https://user-images.githubusercontent.com/8352747/55449346-86224800-55a1-11e9-8a09-3a6103e69949.png)

ninrod commented:
@tam5, be sure to rebase your commits on my latest commit. 

there's a `make emacs` for evil-surround now where you get a clean emacs with just standard evil and standard evil-surround installed and nothing more. 

```sh
make
make emacs
```

then you should be able to reproduce my problem. 

you can also load the tests with `C-x C-f test/evil-surround-test.el`. eval the test buffer and then just `M-x ert` to run the tests locally and interactivelly. 

`make test` should also work locally.

tam5 commented:
@ninrod I've rebased, cleaned up the commits, and looks like the tests pass, functionality works, and no more annoying error message (all as far as I can tell of course).

Please have a look, and see if you can spot any other issues I did not find.

Quick note on the new `make emacs`: it is awesome and was definitely helpful in reproducing your error. For some reason, running the tests in that environment still gave me a different result than when they ran in travis, and the only way I could reproduce was running a fresh docker container. (I am on OS X, but no idea if that is impacting things or if i am just doing something wrong)

ninrod commented:
Yes, I'll take a look later today. For now I think you could complement the tests with html attributes which does not have an equal sign like: 

`<a href="lol.com" prettyLongAngularDirective #anchor1>link</a>`

ninrod commented:
I've reviewed the last commit and all seems good now. Merged. Thanks for your contribution!
